File: structy/array-and-string/most-frequent-char.js
/*
Most Frequent Char 
https://structy.net/problems/premium/most-frequent-char
27 August 2023
David Pratt
*/

const mostFrequentChar = (s) => {
  let obj = {}
  for(let i = 0; i < s.length; i++){
    let key = s[i]
    if(obj[ key ] ) obj[ key ]++;
    else obj[ key ] = 1;
  }
  let vals = Object.values(obj);
  let keys = Object.keys(obj);
  let max = Math.max(...vals);
  let idx = vals.indexOf(max);
  return keys[idx];
};

console.log(mostFrequentChar('bookeeper'));

File: structy/array-and-string/most-frequent-char.py
'''
Most Frequent Char
https://structy.net/problems/premium/most-frequent-char
12 November 2023
David Pratt
'''

def most_frequent_char(s):
    word_length = len(s)
    char_count = {}
    for char in s:
      if char in char_count:
        char_count[char] += 1
      else:
        char_count[char] = 1

    max_key = max(char_count, key=char_count.get)
    return max_key
